src/DAO/scheduleDAO.ts

```diff
@@ -21,6 +33,45 @@ export class ScheduleDAO
     return await scheduleRepository.find({ relations: ['trainer', 'class'] });
   }
 
+  async findByFilters(options: ISearchParams): Promise<Schedule[]> {
+    const scheduleRepository = await this._getRepository(ScheduleEntity);
+    const frDate = options.fromDate;
+    const tilDate = options.untilDate;
+    const train = options.trainer;
+    const time = options.byTime;
+    const srtBy = options.sortBy;
+    const srt = options.sort;
+    const fromDate = new Date(frDate);
+    const untilDate = new Date(tilDate);
```

Comment:
frDate, tilDate, train, time, srtBy, srt, fromDate, untilDate - redundant variables. Can we simplify that? What about one more method in the service layer, to pick up only existing filters?

Reply:
maybe not in service layer(That injects DAO), how about private helper-method in here